### DISCUSSION TITLE
Nodes missing: [Node xxx] is unknown - discarding received command

### DISCUSSION THREAD

**Discussion by ocalvo** [zwave-js/zwave-js#6558] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Even after fix for #6131, I continue to have missing nodes.

What did you expect to happen instead?
I used to have 136 nodes, down to only 63.

1. Unknown, happened after upgrade.

Anything else we should know?
No

### Software versions

Driver (node-zwave-js): ...

Z-Wave JS UI: ...

Home Assistant Z-Wave Integration: ...

Home Assistant Z-Wave JS Addon: ...

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...

### Device information

Manufacturer: ...
Model name: ...
Node ID: ...

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

```
2023-12-15T16:35:44.462Z CNTRLR   [Node 107] is unknown - discarding received command...
2023-12-15T16:35:45.038Z CNTRLR   [Node 107] is unknown - discarding received command...
```
[zwavejs_2023-12-15.log](https://github.com/zwave-js/node-zwave-js/files/13689410/zwavejs_2023-12-15.log)
[zwavejs_current.log](https://github.com/zwave-js/node-zwave-js/files/13689411/zwavejs_current.log)
[zwavejs_logrotate.json](https://github.com/zwave-js/node-zwave-js/files/13689412/zwavejs_logrotate.json)

**Comment by AlCalzone** [comment]
I will try to look at this as soon as I have some time. Can you share a node ID of at least one of the missing nodes?
Also, you could try unplugging and re-plugging the stick in the meantime. Maybe it just needs a restart.

**Comment by ocalvo** [comment]
Node 107 is one that is already part of the network but not recognized by the controller.
I just remove the dongle, turn off yellow hass and replug everything back in. Same behavior.

**Comment by AlCalzone** [comment]
Ok it looks like the controller itself doesn't know these nodes anymore. The list of nodes in the network does not contain Node 107 or any of the other "unknown" nodes:
> nodes in the network:       1, 3, 6, 7, 9, 11, 12, 13, 15, 16, 20, 21, 26, 27, 28, 30, 31, 33, 39, 41, 45, 46, 47, 54, 56, 57, 60, 62, 65, 67, 71, 74, 75, 82, 83, 87, 88, 94, 96, 105, 106, 113, 114, 117, 119, 120, 124, 125, 127, 132, 134, 142, 150, 156, 157, 162, 164, 165, 166, 168, 173, 175, 180, 189, 194, 205, 210, 214, 227

Unless you have an NVM backup from before this happened, I'm afraid you'll have to include all of them again. With a bit of luck they'll keep their node IDs.

I wonder though why this happened. Any chance you pressed this button?
![grafik](https://github.com/zwave-js/node-zwave-js/assets/17641229/bed788f4-c787-4ed1-98f5-c14c0e445a24)

**Reply by ocalvo** [reply]
Not at all. This happened in my home in Costa Rica while I was away.

When I am away I don't touch the network, except for the automatic updates to the addon.

This happened between September and Octuber, and I returned to this home until December and now I am trying to understand what happened.

I think I have a backup of the nvm, but I don't have the heart to restore the backup and loose the rest of nodes.

Is there a way to read data from a backup to inspect theist of nodes in that backup?

I have these backups from Setempber: 
```
2023.09.21\addon_a0d7b954_zwavejs2mqtt_1.15.11\a0d7b954_zwavejs2mqtt\data\store\backups\nvm
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a---         9/15/2023  12:00 AM          49152   NVM_20230915060038.bin
-a---         9/16/2023  12:00 AM          49152   NVM_20230916060036.bin
-a---         9/17/2023  12:00 AM          49152   NVM_20230917060019.bin
-a---         9/18/2023  12:00 AM          49152   NVM_20230918060018.bin
-a---         9/19/2023  12:00 AM          49152   NVM_20230919060021.bin
-a---         9/20/2023  12:00 AM          49152   NVM_20230920060029.bin
-a---         9/21/2023  12:00 AM          49152   NVM_20230921060041.bin
```
Is there a tool I can use to inspect them?

I also have an additional ZWave stick, can I restore to a different dongle?

**Reply by AlCalzone** [reply]
> Is there a tool I can use to inspect them?

There is: https://github.com/zwave-js/node-zwave-js/tree/master/packages/nvmedit#convert-binary-nvm-file-to-json
You can convert it to JSON and look at the list of nodes. In theory, you could also edit the JSON to include the missing nodes and convert it back to `.bin`, then restore it. But you'd have to know some details about the nodes.

> I also have an additional ZWave stick, can I restore to a different dongle?

You can do that too.

**Comment by ocalvo** [comment]
Good news, I found a nvm backup from August.
<img width="751" alt="image" src="https://github.com/zwave-js/node-zwave-js/assets/2091582/c219124d-b7ea-4c17-9543-696a8f14b061">
Was able to recover all devices by plugging the new restored dongle.
Unfortunately, I did lose the name mapping between node id and device name.
Is there a way to get this mapping back by looking at some other part of the backup?
<img width="1045" alt="image" src="https://github.com/zwave-js/node-zwave-js/assets/2091582/957f9baf-0aab-4a67-bc1d-bd656c279385">

**Reply by AlCalzone** [reply]
> Is there a way to get this mapping back by looking at some other part of the backup?

Have you tried restoring the backup on the original dongle? The names are stored in HA and related somehow to the controller hardware. Maybe HA decided that a different controller == a different network.

**Reply by ocalvo** [reply]
Thanks @AlCalzone, I decided to also download a backup from core and use powershell to get the mapping between node id's and the name of the device in homeassistant.

But this time, I renamed every node in zwavejs so that they get stored there.

I have completed this and I am fully restored.

It seems that the upgrade of all 3 range extenders 7 failed for all of them and they drop from the network.

Can that explain why the other 60 nodes also drop from the controller?

**Reply by AlCalzone** [reply]
Normally a failed OTA update shouldn't cause anything - it will simply not get updated.

That amount of nodes disappearing either hints at some issues with the controller memory. Maybe you had a power surge just as the controller was rewriting that part of its flash memory?

The other option would be that lots of devices factory reset themselves, and notified Z-Wave JS of that, which would trigger a removal from the network, but only if the devices did not respond.
However your logs showed that they still send something, so unless they were actually unavailable and Z-Wave JS received a reset notification during that time, this seems unlikely.

**Comment by ocalvo** [comment]
To add to the mistery, this started to happen around the time a firmware upgrade was offered for ZWave extenders, after the upgrade, it seems all extenders became dead. Only one extender survives becuase it was too far away, and the upgrade failed.
<img width="1039" alt="image" src="https://github.com/zwave-js/node-zwave-js/assets/2091582/a2cbd7d2-cb7e-4f34-acbb-902463efc2c8">
Lesson: don't upgrade remotely.

**Comment by duylong** [comment]
Same problem, from time to time I have unknown modules which do not exist in my network. 

I'm the only one with a Zwave network around my house and I don't have any other active modules that aren't included in my network. 

As it is not blocking, I have not identified the problem but I see that I am not alone.

**Comment by Sha73616** [comment]
I'm having the same problem with the message "[Node 075] is unknown - discarding received command..." followed by tons of "DRIVER   Dropping message with invalid payload" which is causing my controller to hang and I have to constantly restart zwave js ui. Node 75 was a motion detector node that I removed from the controller months ago and I cannot locate where it is receiving commands from node 75.
